62/serfer2/pycon_es.py

```diff
@@ -0,0 +1,155 @@
+# coding=utf-8 -*-
+
+import json
+import requests
+import sys
+import pandas
+import matplotlib.pyplot as plt
+
+from scraper import NameScraper
+
+GENDER_API_KEY = 'ERTTRyulCoHJXVhUEF'
+NAMES_CACHE_FNAME = 'names.csv'
+RESULTS_FNAME = 'results.csv'
+PYCON_ES_SOURCES = (
+    {
+        'year': '2018',
+        'url': 'https://2018.es.pycon.org/#schedule',
+        'xpath': '//div[@class="schedule__talk--speaker"]/div/p/text()'
+    }, {
+        'year': '2017',
+        'url': 'https://2017.es.pycon.org/es/schedule/',
+        'xpath': '//div[@class="slot-inner"]/p/strong/text()'
+    }, {
+        'year': '2016',
+        'url': 'http://2016.es.pycon.org/es/schedule/',
+        'xpath': '//div[@class="slot-inner"]/p/strong/text()'
+    }, {
+        'year': '2015',
+        'url': 'http://2015.es.pycon.org/es/schedule/',
+        'xpath': '//div[@class="slot-inner"]/p/strong/text()'
+    }, {
+        'year': '2014',
+        'url': 'http://2014.es.pycon.org/talks',
+        'xpath': '//div[@id="onlytext"]/h1[@class="text-center"]/text()'
+    },
+    {
+        'year': '2013',
+        'url': 'http://2013.es.pycon.org/#agenda',
+        'xpath': '//ul[@class="schedule-list"]/li/div/a[@target="_blank"]/text()'
+    }
+)
+KNOWN_GENDERS = ['male', 'female', 'unknown']
+
+
+def get_gender(name):
+    """
+    Gets name gender from https://gender-api.com
+    """
+    name_data = json.loads(requests.get(f'https://gender-api.com/get?name={name}&key={GENDER_API_KEY}').content)
+    return name_data.get('gender', '')
+
+
+def load_names_cache():
+    """
+    Read local files with names by gender
+    """
+    names = dict()
+    with open(NAMES_CACHE_FNAME, 'r') as fi:
+        lines = fi.readlines()
+        for line in lines:
+            line = line.strip()
+            if not line:
+                continue
+            name, gender = line.split(';')
+            names[name] = gender
```

Comment:
can be combined in one list comprehension, even the assignment names = can be combined with list comprehension using `dict()` method with tuples. So something like `names = dict(((l.split(";")[0], l.split(";")[1]) for l in fi.readlines() if l))`